tests/benchmarks/conftest.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+
+@pytest.yield_fixture(scope="function")
+def aio_benchmark(benchmark):
+    import asyncio
+    import threading
+
+    class Sync2Async:
+        def __init__(self, coro, *args, **kwargs):
+            self.coro = coro
+            self.args = args
+            self.kwargs = kwargs
+            self.custom_loop = None
+            self.thread = None
+
+        def start_background_loop(self) -> None:
+            asyncio.set_event_loop(self.custom_loop)
+            self.custom_loop.run_forever()
+
+        def __call__(self):
+            evloop = None
+            awaitable = self.coro(*self.args, **self.kwargs)
+            try:
+                evloop = asyncio.get_running_loop()
+            except Exception:
+                pass
+            if evloop is None:
+                return asyncio.run(awaitable)
+            else:
+                if (
+                    not self.custom_loop
+                    or not self.thread
+                    or not self.thread.is_alive()
+                ):
+                    self.custom_loop = asyncio.new_event_loop()
+                    self.thread = threading.Thread(
+                        target=self.start_background_loop, daemon=True
+                    )
+                    self.thread.start()
+
+                return asyncio.run_coroutine_threadsafe(
+                    awaitable, self.custom_loop
+                ).result()
```

Comment:
can we use the one from asgiref? https://github.com/django/asgiref#function-wrappers

Reply:
wow, im actually just replacing this with asgiref as we speak....